lib/os/mempool.c

```diff
@@ -328,8 +328,10 @@ void *sys_mem_pool_alloc(struct sys_mem_pool *p, size_t size)
 	struct sys_mem_pool_block *blk;
 	uint32_t level, block;
 	char *ret;
+	int err;
 
-	sys_mutex_lock(&p->mutex, K_FOREVER);
+	err = sys_mutex_lock(&p->mutex, K_FOREVER);
+	__ASSERT(err == 0, "sys_mutex_lock failed with %d", err);
```

Comment:
should consider using CHECKIF here:

```
err = sys_mutex_lock(&p->mutex, K_FOREVER);
CHECKIF(err != 0) {
	ret = NULL;
	goto out;
}
```
also, err as a variable name might be misleading, the function might return a non error return code, you could use ret but it is already used :)

Reply:
Changed both instances to use CHECKIF and changed variable names from err to lock_ret. Would that variable name be more clear?